src/Host/Broker/Impl/Startup/Program.cs

```diff
@@ -57,7 +61,38 @@ public class Program {
 
             var tlsConfig = new TlsConfiguration(_logger, _securityOptions);
             var httpsOptions = tlsConfig.GetHttpsOptions(Configuration);
-            CreateWebHost(httpsOptions).Run();
+
+            Uri uri = GetServerUri(Configuration);
+            if (IsPortAvailable(uri?.Port)) {
```

Comment:
@int19h  I am running into a issue where the logger is disposed before i can log the port failure, in the corner case. An exception in webHost.Run() (UvException with EADDRINUSE) results in logger getting disposed and by the time we get to logging the message there is no file logger present. Any suggestions on how to fix that?